packages/router-plugin/src/core/config.ts

```diff
@@ -3,9 +3,59 @@ import {
   configSchema as generatorConfigSchema,
   getConfig as getGeneratorConfig,
 } from '@tanstack/router-generator'
+import { splitRouteIdentNodes } from './constants'
+import type { CodeSplitGroupings } from './constants'
+
+export const splitGroupingsSchema = z
+  .array(z.array(z.enum(splitRouteIdentNodes)), {
+    message:
+      "  Must be an Array of Arrays containing the split groupings. i.e. [['component'], ['pendingComponent'], ['errorComponent', 'notFoundComponent']]",
+  })
+  .superRefine((val, ctx) => {
+    const flattened = val.flat()
+    const unique = [...new Set(flattened)]
+
+    // Elements must be unique,
+    // ie. this shouldn't be allows [['component'], ['component', 'loader']]
+    if (unique.length !== flattened.length) {
+      ctx.addIssue({
+        code: 'custom',
+        message:
+          "  Split groupings must be unique and not repeated. i.e. i.e. [['component'], ['pendingComponent'], ['errorComponent', 'notFoundComponent']]." +
+          `\n  You input was: ${JSON.stringify(val)}.`,
+      })
+    }
+  })
+
+export type CodeSplittingOptions = {
+  /**
+   * Use this function to programmatically control the code splitting behaviour
+   * based on the `routeId` for each route.
+   *
+   * If you just need to change the default behaviour, you can use the `defaultBehaviour` option.
+   * @param params
+   */
+  splitBehaviour?: (params: {
```

Comment:
probably best to use US spelling? `behavior` without `u`

Reply:
Could you apply that change while doing the type-safe `routeId` stuff?